app_utils/sections/experiment.py

```diff
@@ -1900,20 +1905,33 @@ def load_cfg_model_tokenizer(experiment_path, merge=False, device="cuda:0"):
         cfg.architecture.backbone_dtype = "float16"
         cfg.architecture.pretrained = True
 
-    with torch.device(device):
+    with torch.device(cfg.environment._device):
         model = cfg.architecture.model_class(cfg)
         cfg.architecture.pretrained_weights = os.path.join(
             experiment_path, "checkpoint.pth"
         )
         load_checkpoint(cfg, model, strict=False)
 
-        if merge and cfg.training.lora:
-            # merges the LoRa layers into the base model.
-            # This is needed if one wants to use the base model as a standalone model.
-            logger.info("Merging LORA layers with base model.")
-            model.backbone = model.backbone.merge_and_unload()
+    if merge and cfg.training.lora:
+        # merges the LoRa layers into the base model.
+        # This is needed if one wants to use the base model as a standalone model.
+        logger.info("Merging LORA layers with base model.")
+        model.backbone = model.backbone.merge_and_unload()
+
+    if device == "cpu_shard":
+        from accelerate import dispatch_model
```

Comment:
Can we import thid on top level?

Reply:
Commonly, it is top level imports. I would restrict local imports to specific cases where it solves some import issue (delayed import for slow packages/side effects when importing/avoiding circular imports/ ...).